src/bootstrap-lib/keyval.c

```diff
@@ -61,7 +61,7 @@ int bootstrap_keyval_parse(char *path) {
             if ( strncmp(bootdef_key, "#", 1) != 0 ) {
                 char *bootdef_value;
 
-                if ( ( bootdef_value = strtok(NULL, "\n") ) != NULL ) {
+                if ( ( bootdef_value = strtok(NULL, "#\n") ) != NULL ) {
```

Comment:
I think it will take a bit more logic to do this right. I would propose adding a chomp_comments() function in util.c, and using that here.

Thoughts?